src/helpers.js

```diff
@@ -66,17 +66,25 @@ export function isValidWhere (where) {
   return [false]
 }
 
-export function runQuery (instance, query, needResponse) {
+export function runQuery ({ instance, name, query, needResponse }) {
   if (util.isFunction(instance.verbose)) {
     instance.verbose(query.toString())
   }
 
   if (instance.isNative) {
-    if (needResponse) return query
-    return query.then(res => {
-      if (util.isNumber(res)) return res
-      return res ? res.length : 0
-    })
+    if (name && /^insert/i.test(query.toString())) {
```

Comment:
Well, I got what I need with `instance.knex.queryBuilder()._method`. Both clients. Is that ok?